### PR TITLE
Proposal to add an abbreviation section ommitting 'well known' ones b…

### PR DESCRIPTION
**Pull request by taddhar**
…ut yet for a large audience.

### CODE REVIEW

File: draft-ietf-tls-esni.md [hunk 1]
```diff
@@ -116,6 +116,68 @@ document are to be interpreted as described in BCP 14 {{RFC2119}} {{!RFC8174}}
 when, and only when, they appear in all capitals, as shown here. All TLS
 notation comes from {{RFC8446, Section 3}}.
 
+# Abbreviations
```
Comment (roelfdutoit) [inline draft-ietf-tls-esni.md:119]: I don't recall ever seeing this in the RFCs/drafts I have studied.  All the abbreviations below come from normative and informative references listed at the end of the draft.  The specification is written with implementers in mind, which should already be familiar with the terminology.  This PR will likely be rejected by the TLS WG.

File: draft-ietf-tls-esni.md [hunk 2]
```diff
@@ -116,6 +116,68 @@ document are to be interpreted as described in BCP 14 {{RFC2119}} {{!RFC8174}}
...
+AEAD
+: Authenticated Encryption with Associated Data
+
+ALPN
+: Application-Layer Protocol Negotiation
+
+CA
+: Certificate Authority
+
+CH
```
Comment (roelfdutoit) [inline draft-ietf-tls-esni.md:136]: `CH` + `HRR` + `EE`, as mentioned in the "Update of the TLS ExtensionType Registry" section, are TLS handshake messages defined in RFC8446.
CH = Client Hello
HRR = Hello Retry Request
EE = Encrypted Extensions

File: draft-ietf-tls-esni.md (hunk 1 above)
Comment (taddhar) [reply draft-ietf-tls-esni.md:119]: Well, this PR may be rejected by the TLS WG but that doesn't mean that I agree with a number of things:
1) Whilst audience is for implementers in mind, I disagree with that this is the only population to read such a specification. 

Which is the whole drama of the IETF, there are operational people who need to read these specs too and in particular security operational people. I am thinking about people who will want to establish policies and whatever else that we can't pre-empt. 

2) I personnaly DON'T like an approach which is implicit and where even the minimum of expanding an acronym is not even respected. That is a BASIC reader's respect that people like me hate because this is adding barrier to entry to audiences who precisely need explicit. I don't understand the principle of writing text that forces your brain to have to constantly caclulate and overload your memory on a protocol which IS complex, new relying on a LOT of dependencies that then you have to dig in into other documents. Etc.

3) I note that the Abbreviation list of the IETF which I think is a very good initiative was not updated since 2021! This is bad. 

So I can compromise with no Abbreviation section att he condition to check that each abbreviation is properly expanded each time of its first appearance and not that I have to guess, search at the end or have to read the whole document to go back and forth. 

Abbreviations is a very good practice in MANY other standards (ITU, ISO, etc. etc.) and I don't see what it harms to put it in.

File: draft-ietf-tls-esni.md (hunk 2 above)
Comment (taddhar) [reply draft-ietf-tls-esni.md:136]: And here it is, that I made a mistake on HRR because it collides too with HTTP RR so this is a perfect case that for a non TLS implementor it is hard to follow and just adds extra confusion.
